charts/external-dns/values.yaml

```
@@ -180,7 +180,20 @@ txtSuffix: ""
 
 domainFilters: []
 
-provider: aws
+provider:
+  # When the provider is not in-tree, it will add a sidecar, using provider.images matching package
+  name: aws
+  version: latest
+  env: []
+  args: []
+  resources: {}
+  securityContext: {}
+  images:
+    adguard: ghcr.io/muhlba91/external-dns-provider-adguard
+    glesys: ghcr.io/glesys/external-dns-glesys
+    hetzner: ghcr.io/mconfalonieri/external-dns-hetzner-webhook
+    ionos: ghcr.io/ionos-cloud/external-dns-ionos-webhook
+    stackit: ghcr.io/stackitcloud/external-dns-stackit-webhook
```

```suggestion
provider:
  # When the provider is not in-tree, it will add a sidecar using the webhook values
  name: aws
  
  webhook:
    image:
      repository:
      tag:
      pullPolicy: IfNotPresent
    env: []
    args: []
    resources: {}
    securityContext: {}
```
I think nesting the webhook specific values under a `webhook` key makes it easier to understand. it also makes sense to mirror the equivalent chart values for the main container.

I'm also not a fan of specifying images that aren't at least Semver tagged. As these other images are out of tree their releases aren't going to line up with ExtenalDNS and as such I don't think they should be embedded in the chart. For context we don't embed support most args as strongly typed values in the chart so we have smaller maintenance surface area. Maybe we could set the default webhook image as ExternalDNS?

---

The primary thing being configured here is the use of a DNS provider that is supplied as an image and which runs as a sidecar. "webhook" is an implementation detail of how those two containers communicate and is not necessary to or part of the admin's understanding of what they need to configure.